tests/test_connector.py

```diff
@@ -1709,67 +1709,109 @@ async def test_tcp_connector_clear_dns_cache_bad_args(
+async def test___get_ssl_context3() -> None:
     ctx = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
     conn = aiohttp.TCPConnector(ssl=ctx)
     req = mock.Mock()
     req.is_ssl.return_value = True
     req.ssl = True
-    assert conn._get_ssl_context(req) is ctx
+    assert await conn._get_ssl_context(req) is ctx
 
 
-async def test___get_ssl_context4(loop: asyncio.AbstractEventLoop) -> None:
+async def test___get_ssl_context4() -> None:
     ctx = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
     conn = aiohttp.TCPConnector(ssl=ctx)
     req = mock.Mock()
     req.is_ssl.return_value = True
     req.ssl = False
-    assert conn._get_ssl_context(req) is conn._make_ssl_context(False)
+    assert await conn._get_ssl_context(req) is await conn._make_or_get_ssl_context(
+        False
+    )
 
 
-async def test___get_ssl_context5(loop: asyncio.AbstractEventLoop) -> None:
+async def test___get_ssl_context5() -> None:
     ctx = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
     conn = aiohttp.TCPConnector(ssl=ctx)
     req = mock.Mock()
     req.is_ssl.return_value = True
     req.ssl = aiohttp.Fingerprint(hashlib.sha256(b"1").digest())
-    assert conn._get_ssl_context(req) is conn._make_ssl_context(False)
+    assert await conn._get_ssl_context(req) is await conn._make_or_get_ssl_context(
+        False
+    )
 
 
-async def test___get_ssl_context6(loop: asyncio.AbstractEventLoop) -> None:
+async def test___get_ssl_context6() -> None:
     conn = aiohttp.TCPConnector()
     req = mock.Mock()
     req.is_ssl.return_value = True
     req.ssl = True
-    assert conn._get_ssl_context(req) is conn._make_ssl_context(True)
+    assert await conn._get_ssl_context(req) is await conn._make_or_get_ssl_context(True)
+
+
+async def test_ssl_context_once() -> None:
+    """Test the ssl context is created only once and shared between connectors."""
+    conn1 = aiohttp.TCPConnector()
+    conn2 = aiohttp.TCPConnector()
+    conn3 = aiohttp.TCPConnector()
+
+    req = mock.Mock()
+    req.is_ssl.return_value = True
+    req.ssl = True
+    assert await conn1._get_ssl_context(req) is await conn1._make_or_get_ssl_context(
+        True
+    )
+    assert await conn2._get_ssl_context(req) is await conn1._make_or_get_ssl_context(
+        True
+    )
+    assert await conn3._get_ssl_context(req) is await conn1._make_or_get_ssl_context(
+        True
+    )
+    assert conn1._made_ssl_context is conn2._made_ssl_context is conn3._made_ssl_context
+    assert True in conn1._made_ssl_context
+
+
+@pytest.mark.parametrize("exception", [OSError, ssl.SSLError, asyncio.CancelledError])
+async def test_ssl_context_creation_raises(exception: BaseException) -> None:
+    """Test that we try again if SSLContext creation fails the first time."""
+    conn = aiohttp.TCPConnector()
+    conn._made_ssl_context.clear()
+
+    with mock.patch.object(
+        conn, "_make_ssl_context", side_effect=exception
+    ), pytest.raises(  # type: ignore[call-overload]
```

Comment:
Commit says this is missing BaseException on the overload?
Code appears to be correct: https://github.com/pytest-dev/pytest/blame/main/src/_pytest/python_api.py#L778

Reply:
It sure does look right there.

Maybe we have an old version?

I can't find the CI run so I started another one https://github.com/aio-libs/aiohttp/pull/8676

Reply:
https://github.com/aio-libs/aiohttp/actions/runs/10332753102/job/28604325639?pr=8676

Reply:
Just a silly mistake on my part. I forgot it needed to be wrapped in `type[]`